server/utils/assessmentUtils.ts

```diff
@@ -200,28 +200,54 @@ const getTaskResponsesAsSummaryListItems = (
   task: Task,
   application: ApprovedPremisesApplication,
 ): Array<SummaryListItem> => {
-  const items: Array<SummaryListItem> = []
-
   if (!application.data[task.id]) {
-    return items
+    return []
   }
 
   const pageNames = Object.keys(application.data[task.id])
 
-  pageNames.forEach(pageName => {
+  return pageNames.reduce((prev, pageName) => {
     const response = getResponseForPage(application, task.id, pageName)
-    Object.keys(response).forEach(key => {
-      const value =
-        typeof response[key] === 'string' || response[key] instanceof String
-          ? ({ text: response[key] } as TextItem)
-          : ({ html: embeddedSummaryListItem(response[key] as Array<Record<string, unknown>>) } as HtmlItem)
-
-      items.push({
-        key: {
-          text: key,
-        },
-        value,
-      })
+
+    if (pageName === 'attach-documents') {
+      const summaryListItems = getAttatchDocumentsSummaryListItems(application)
+      return [...prev, ...summaryListItems]
+    }
+
+    const summaryListItems = getGenericSummaryListItems(response)
+    return [...prev, ...summaryListItems]
```

Comment:
Could we do a conditional assignment here?

```suggestion
      const summaryListItems = pageName === 'attach-documents' ? getAttatchDocumentsSummaryListItems(application) : getGenericSummaryListItems(response)
    
    return [...prev, ...summaryListItems]
```